### PR TITLE
Move external provider and components into theme

### PR DESCRIPTION
**Pull request by busypeoples**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -200,13 +198,16 @@ export class SlideDeck extends React.Component {
...
-      components,
-      Provider,
       width,
       height
     } = this.props
     const { index, length, mode, step} = this.state
 
+    const {
+      components = {},
+      Provider = DefaultProvider
```
Comment (jxnblk) [inline src/index.js]: Nice!

File: src/entry.js [hunk 2]
```diff
@@ -4,16 +4,14 @@ import SlideDeck from './index'
 
 const mod = require(DOC_FILENAME)
 const slides = mod.default
-const { theme, components, Provider } = mod
```
Comment (jxnblk) [inline src/entry.js]: This looks like it'd mean a breaking change – what do you think of keeping backwards compatibility even if it's not documented?

Comment (busypeoples) [reply src/entry.js]: See what you mean. Then let's keep it.

**Comment by busypeoples** [comment]
No idea why this doesn't pass. Double checked that `DefaultProvider` is not declared twice. This is interesting.

**Comment by jxnblk** [comment]
I suspect it's because of this with the merge: https://github.com/jxnblk/mdx-deck/blob/master/src/index.js#L8

**Comment by jxnblk** [comment]
Awesome, LGTM!

Fixes #63

**Pull request merged**